apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/SimpleManagedTracer.cs

```diff
@@ -210,23 +249,20 @@ public void SetStackTrace(StackTrace stackTrace)
             var currentStack = TraceStack;
             CheckStackNotEmpty(currentStack);
 
-            AnnotateSpan(currentStack.Peek(), TraceLabels.FromStackTrace(stackTrace));
+            AnnotateSpan(currentStack.Peek().TraceSpan, TraceLabels.FromStackTrace(stackTrace));
         }
 
         /// <inheritdoc />
-        public string GetCurrentTraceId()
-        {
-            return _trace.TraceId;
-        }
-
+        public string GetCurrentTraceId() =>_trace.TraceId;
+        
         /// <inheritdoc />
         public ulong? GetCurrentSpanId() => GetCurrentSpanId(TraceStack);
 
         /// <summary>
         /// Gets the current span id of the specified stack or null if none exists.
         /// </summary>
-        private ulong? GetCurrentSpanId(ImmutableStack<TraceSpan> traceStack)
-            => traceStack.IsEmpty ? _rootSpanParentId : traceStack.Peek().SpanId;
+        private ulong? GetCurrentSpanId(ImmutableStack<Span> traceStack) =>
```

Comment:
This will need to pop off all disposed spans as well: when annotating spans, we don't want to annotate one ended from another thread. Similarly, then starting a span we don't want the new one to be parented to a disposed span.